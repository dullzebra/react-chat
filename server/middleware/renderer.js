import React from 'react';
import ReactDOMServer from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import { SheetsRegistry } from 'react-jss/lib/jss';
import JssProvider from 'react-jss/lib/JssProvider';
import { createGenerateClassName } from 'material-ui/styles';

import configureStore from '../../src/store';
import App from '../../src/components/App';

const sheetsRegistry = new SheetsRegistry();
const generateClassName = createGenerateClassName();

const path = require('path');
const fs = require('fs');

export default (req, res) => {
  // point to the html file created by CRA's build tool
  const filePath = path.resolve(__dirname, '..', '..', 'build', 'index.html');
  const context = {};
  const store = configureStore();

  fs.readFile(filePath, 'utf8', (err, htmlData) => {
    if (err) {
      console.error('err', err);
      return res.status(404).end();
    }

    if (context.url) {
      res.redirect(context.url);
      return;
    }

    // render the app as a string
    /* eslint-disable-next-line */
    const html = ReactDOMServer.renderToString(
      <JssProvider registry={sheetsRegistry} generateClassName={generateClassName}>
        <Provider store={store}>
          <StaticRouter location={req.url} context={context}>
            <App />
          </StaticRouter>
        </Provider>
      </JssProvider>);

    const css = sheetsRegistry.toString();

    // inject the rendered app into our html and send it
    return res.end(htmlData.replace(
      '<div id="root"></div>',
      `<div id="root">${html}</div>
    <script>window.INITIAL_STATE = ${JSON.stringify(store.getState()).replace(
    /</g,
    '\\u003c',
  )}</script>
    <style id="jss-server-side">${css}</style>
    `,
    ));
  });
};
