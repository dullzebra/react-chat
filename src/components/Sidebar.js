import React from 'react';
import { withStyles } from 'material-ui/styles';
import Drawer from 'material-ui/Drawer';
import Divider from 'material-ui/Divider';
import ChatList from './ChatList';
import SidebarFilter from './SidebarFilter';
import SidebarActions from './SidebarActions';
import * as style from '../utils/constants';

const styles = theme => ({
  drawerPaper: {
    position: 'relative',
    width: style.drawerWidth,
    overflow: 'hidden'
  }
});


const Sidebar = ({ classes, chats }) => (
  <Drawer variant="permanent" classes={{ paper: classes.drawerPaper }}>
    <SidebarFilter />
    <Divider />
    <ChatList chats={chats} />
    <SidebarActions />
  </Drawer>
);

export default withStyles(styles)(Sidebar);
