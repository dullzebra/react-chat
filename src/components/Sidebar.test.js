/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Sidebar from './Sidebar';

jest.mock('./ChatList', () => () => 'ChatList');
jest.mock('./SidebarActions', () => () => 'SidebarActions');

const mockData = {
  chats: {
    active: {},
    allChats: [],
    myChats: [],
  },
  createChat: jest.fn(),
  isConnected: true,
};

describe('<Sidebar />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Sidebar {...mockData} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Sidebar {...mockData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
