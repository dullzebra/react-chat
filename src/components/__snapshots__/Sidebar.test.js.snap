// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> renders correctly 1`] = `
<div
  className="MuiDrawer-docked-41"
>
  <div
    className="MuiPaper-root-52 MuiPaper-shadow0-54 MuiDrawer-paper-42 Sidebar-drawerPaper-40 MuiDrawer-paperAnchorLeft-43 MuiDrawer-paperAnchorDockedLeft-47"
  >
    ChatList
    SidebarActions
  </div>
</div>
`;
